### DISCUSSION TITLE
How to use rbatis to implement the inner join query ?

### DISCUSSION THREAD

**Discussion by cnzf1** [rbatis/rbatis#265] [General]
Now, I want to query some data from two table,  and I often user query! macro to query the table generally. For example:

#[crud_table]
#[derive(Serialize, Deserialize, Clone, Debug)]
pub struct Storage {
    pub id: Option<u32>,
    pub nip: Option<String>,
    pub uuid: Option<String>,
}

  let rb = db::get_rbatis().await?;
    let r = rb
        .fetch_by_column::<Option<Storage>, _>("id", id)
        .await?;

Ok(r.unwrap().uuid)

If I want query the data , the sql like 'select b.name from a inner join b on a.bid=b.id, where a.id=111'.  What should I do ?

**Comment by easykoala** [comment]
can use this:

#[py_sql("select b.name from a inner join b on a.bid=b.id, where a.id=111)]
 async fn py_sql_tx(rb: &Rbatis) -> Vec<Storage > { impled!() }

**Comment by zhuxiujia** [comment]
> can use this:
> 
> #[py_sql("select b.name from a inner join b on a.bid=b.id, where a.id=111)] async fn py_sql_tx(rb: &Rbatis) -> Vec { impled!() }

yes,  
```
#[py_sql("select b.name from a inner join b on a.bid=b.id, where a.id=111)]
async fn py_sql_tx(rb: &Rbatis) -> Vec<rbson::Bson> { impled!() }
```
